paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentSheetScreen.kt

```diff
@@ -409,28 +410,35 @@ internal fun Wallet(
     onLinkPressed: () -> Unit,
     dividerSpacing: Dp,
     modifier: Modifier = Modifier,
-    cardBrandFilter: CardBrandFilter
+    cardBrandFilter: CardBrandFilter,
+    isCompleteFlow: Boolean
 ) {
     val padding = StripeTheme.getOuterFormInsets()
+    var hasWalletButtons = false
 
     Column(modifier = modifier.padding(padding)) {
-        state.googlePay?.let { googlePay ->
-            GooglePayButton(
-                state = PrimaryButton.State.Ready,
-                allowCreditCards = googlePay.allowCreditCards,
-                buttonType = googlePay.buttonType,
-                billingAddressParameters = googlePay.billingAddressParameters,
-                isEnabled = state.buttonsEnabled,
-                onPressed = onGooglePayPressed,
-                cardBrandFilter = cardBrandFilter
-            )
+        val showGooglePayInWallet = isCompleteFlow // Only complete flows show Google Pay in wallet
```

Comment:
Can we have this logic in the `WalletsState`?

Reply:
replaced a few boolean flags in favor of a list of wallets that are allowed to show on the header.